packages/task/src/browser/task-service.ts

```diff
@@ -409,6 +410,73 @@ export class TaskService implements TaskConfigurationClient {
     }
 
     async runTask(task: TaskConfiguration, option?: RunTaskOption): Promise<TaskInfo | undefined> {
+        const runningTasksInfo: TaskInfo[] = await this.getRunningTasks();
+
+        // check if the task is active
+        const matchedRunningTaskInfo = runningTasksInfo.find(taskInfo => {
+            const taskConfig = taskInfo.config;
+            return this.taskDefinitionRegistry.compareTasks(taskConfig, task);
+        });
+        if (matchedRunningTaskInfo) { // the task is active
+            const taskName = this.taskNameResolver.resolve(task);
+            const terminalId = matchedRunningTaskInfo.terminalId;
+            if (terminalId) {
+                const terminal = this.terminalService.getById(this.getTerminalWidgetId(terminalId));
+                if (terminal) {
+                    this.shell.activateWidget(terminal.id); // make the terminal visible and assign focus
+                }
+            }
+            this.messageService.info(`The task \'${taskName}\' is already active`, 'Terminate Task', 'Restart Task').then(async actionTask => {
+                if (actionTask === 'Terminate Task') {
+                    this.terminateTask(matchedRunningTaskInfo);
+                } else if (actionTask === 'Restart Task') {
+                    this.restartTask(matchedRunningTaskInfo, option);
+                }
+            });
+            return;
+        } else { // run task as the task is not active
+            return this.doRunTask(task, option);
+        }
+    }
+
+    /**
+     * Terminates a task that is actively running, and closes the terminal (if any) that the actively running task is attached to.
+     * @param activeTaskInfo the TaskInfo of the task that is actively running
+     */
+    protected async terminateTask(activeTaskInfo: TaskInfo): Promise<void> {
+        const taskId = activeTaskInfo.taskId;
+        this.kill(taskId);
+        const terminalId = activeTaskInfo.terminalId;
+        if (terminalId) {
+            const terminal = this.terminalService.getById(this.getTerminalWidgetId(terminalId));
+            if (terminal) {
+                const parent = terminal.parent;
+                terminal.close();
+                const hasOtherTerminals = this.terminalService.all.length > 0;
+                if (!hasOtherTerminals && parent && parent.id === BOTTOM_AREA_ID) {
+                    parent.hide();
```

Comment:
Do we need it at all? Should not we check at least that there is no other widgets in the bottom area interested for a user?

Reply:
I agree with you Anton. Like I described [here](https://github.com/eclipse-theia/theia/pull/6668#issuecomment-561790788), I am complying with VS code behavior although I dislike it. 
I am not a product manager so my words of "I like it" or "i dislike it" don't count :)

Reply:
in my opinion, 
- we should not close the terminal after the task is terminated. WHY vscode assumes the terminal output is not interesting to developers after terminating the task?
- we should NOT hide the panel, as there might be other widgets  in the panel that I want to keep watching. 

again, I have no idea why vs code does it this way. if most reviewers agree with what I proposed. I will update the code and make it work the Theia way. @akosyakov @RomanNikitenko @lmcbout @vince-fugnitto 

Please share your opinions and insight with me. Thank you !

Reply:
Ooops I am not sure why my comment disappeared so I am going to post it again.

In my opinion, 
- the terminal should NOT be closed after terminating the task, because the task output could be interesting to the user, and
- the bottom panel should NOT be hidden, because there could be other widgets docked. 

If most reviewers here agree with this proposal, I will update my code to make the task termination the Theia way. 

Could you please share your insights ? @akosyakov @lmcbout @RomanNikitenko @vince-fugnitto

Reply:
@elaihau I think you should go with your gut and keep the terminal and bottom area displayed :)

I think each of us (@lmcbout, yourself and I) all agree that VS Code's behavior is odd.

In my opinion:
- the bottom panel might have multiple different widgets present (ex: `memory view`), and hiding it when a task is incorrect. There is value in displaying the bottom area.
- the termination of a task may still hold valuable information and output that a user might care about
- I think the behavior of closing an entire area of the application is annoying. I think it should be up to the user to manually make that decision and not for the application to assume it should be closed.

Reply:
> In my opinion,
> 
> * the terminal should NOT be closed after terminating the task, because the task output could be interesting to the user, and
> * the bottom panel should NOT be hidden, because there could be other widgets docked.
> 
> Again, I don't know WHY vscode closes the terminal and hides the panel. I dislike it.
> 
> If most reviewers here agree with this proposal, I will update my code to make the task termination the Theia way.
> 
> Could you please share your insights ? @akosyakov @lmcbout @RomanNikitenko @vince-fugnitto

I commented about the different behavior from VS Code, because of alignment theia behavior with VS Code behavior, but I'm totally agree with @elaihau and as a developer I prefer do **not** hide the bottom area and do **not** close terminal after terminating a task. User has opportunity to hide bottom panel using hotkey.

Reply:
let's remove it for now then and wait for real issues, it also makes code simpler

Reply:
I agree with @vince-fugnitto and @elaihau : The behavior to remove the terminal when task terminates is odd. I prefer keeping the terminal open when the task is terminated. One thing, we need to be careful because the terminal seems not be registered after, so if we resize the view after termination, I can see a trace in the back-end side, not harmful, but not necessary to clutter the console with the error either.  So my opinion would be to be able to see the terminal view even after task termination, not hide it.